docs/deployment.md

```diff
@@ -0,0 +1,483 @@
+KubeVirt Deployment
+=
+
+# Preface
+When deploying [KubeVirt](https://kubevirt.io/) onto a [Kubernetes](https://kubernetes.io/) cluster, we deploy the following:
+
+- `virt-operator` and its dependencies
+- KubeVirt CustomResource
+
+This is done using the [kubectl](https://kubernetes.io/docs/reference/kubectl/) command line tool or programmatically.
+
+```shell
+# Point at latest stable release
+export RELEASE=$(curl https://storage.googleapis.com/kubevirt-prow/release/kubevirt/kubevirt/stable.txt)
+# Deploy the virt-operator
+kubectl apply --filename https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/kubevirt-operator.yaml
+# Create the KubeVirt CR (instance deployment request) which triggers the actual installation
+kubectl apply --filename https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/kubevirt-cr.yaml
+# Wait until all KubeVirt components are up
+kubectl --namespace=kubevirt wait kubevirt kubevirt --for condition=Available
+```
+
+# Virt Operator Deployment
+
+`virt-operator` is the component responsible for deploying the components of the KubeVirt project.
+
+## Namespace
+A dedicated [Namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) is created for KubeVirt's components.
+
+```shell
+kubectl get namespace kubevirt -o yaml
+```
+
+## KubeVirt CustomResourceDefinition
+KubeVirt extends Kubernetes using [CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/) (CRDs).
+
+The first deployed CRD is `kubevirts.kubevirt.io` which allows us to later create the `KubeVirt` CustomResource.
+
+```shell
+kubectl get customresourcedefinitions.apiextensions.k8s.io kubevirts.kubevirt.io -o yaml
```

Comment:
For readability, I think we should drop all the *.k8s.io suffixes, like:

```suggestion
kubectl get customresourcedefinitions kubevirts.kubevirt.io -o yaml
```

Reply:
Removed most of the suffixes, except for the more advanced resources:
- priorityclasses.scheduling.k8s.io
- validatingwebhookconfigurations.admissionregistration.k8s.io
- mutatingwebhookconfigurations.admissionregistration.k8s.io
- apiservices.apiregistration.k8s.io